src/domain/datasets/domain/src/repos/dataset_dependency_repository.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::pin::Pin;
+
+use internal_error::InternalError;
+use opendatafabric::DatasetID;
+use thiserror::Error;
+use tokio_stream::Stream;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+pub trait DatasetDependencyRepository: Send + Sync {
+    async fn stores_any_dependencies(&self) -> Result<bool, InternalError>;
```

Comment:
How about replacing this with `edges_count()` as a more versatile method.

Reply:
We only need to know if the repository is empty.
Is empty != Count edges from performance point of view.
If we will need to know the number of edges, we will add a separate API.

Somehow, standard containers have both `is_empty` and `len`.
Especially with more advanced structures like linked lists and trees this  this makes a lot more sense, right?
So, it's the same, I'm hoping to use cheaper queries for quick tests.